### PR TITLE
Add optional extra attributes to model and add NIH edge attribute models

### PR DESCRIPTION
**Pull request by flibbertigibbet**
- Sets optional additional attributes on StreetEdge (for use in weighting later)
- Defines expected fields and options from NIH segments shapefile
- Reads in attributes from NIH segments shapefile and sets the extra attributes on the StreetEdges

### CODE REVIEW

**Comment by CloudNiner** [comment]
Is it worth renaming the NihOptions classes to something more generic, in the interest of possibly submitting a PR upstream? Setting properties on a giving vertex/edge seems like a useful feature, and most of the ones you defined here are not necessarily specific to NIH.

File: src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java [hunk 1]
```diff
@@ -349,10 +391,94 @@ public void buildGraph(Graph graph, HashMap<Class<?>, Object> extra) {
...
+            }
+
+            if (val.isEmpty()) {
+                LOG.warn("Skipping setting an empty value for option {}.  Was this intentional?", option.getFieldName());
+            } else {
+
+                // TODO: some better way to correct for left/right values encoded in a single column?
+                if (val.equals("Right")) {
+                    if (isRightSide) {
+                        val = "Both";
```
Comment (CloudNiner) [inline src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java:465]: I'm not sure what the "Both" sprinkled throughout this PR means? Is it insinuating that the attribute is on both sides of the street? But, then here, we set to both if its only right side.

Is idea what the TODO above is referring to?

**Comment by flibbertigibbet** [comment]
That's why it's all genericized, with the NIH stuff in its own sub-package.  To implement with a different field set, one would simply have to create a new fields enum (like NihOptions) and classes for the field options.  This could usefully go upstream with NihOptions and the related field definitions omitted.

File: src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java [hunk 2]
```diff
@@ -340,6 +379,9 @@ public void buildGraph(Graph graph, HashMap<Class<?>, Object> extra) {
             ///////////////////////////////////////////////////////////////////////
 
             // TODO: Set edge properties
```
Comment (CloudNiner) [inline src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java]: Remove todo?

File: src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java [hunk 3]
```diff
@@ -349,10 +391,94 @@ public void buildGraph(Graph graph, HashMap<Class<?>, Object> extra) {
         }
     }
 
+    private EnumMap<NihOption, ArrayList<StringAttributeFeatureConverter>> getFieldConverters() {
```
Comment (CloudNiner) [inline src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java:398]: Short docstrings on this and setProperties below might be helpful

File: src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java (hunk 1 above)
Comment (flibbertigibbet) [inline src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java:465]: Annoyingly, some of our fields with right and left values are defined as two separate fields, and some are defined as one field with the values of "Left", "Right", or "Both", but the street segments are directional, so it's only going to be flagged on or off.  So to simplify, I've normalized all the option values like that, for now, to be only either "Both" or "None".

**Comment by CloudNiner** [comment]
LGTM at a high level, my java is a bit rusty though, so buyer beware.

File: src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java (hunk 2 above)
Comment (flibbertigibbet) [inline src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java]: Yup

File: src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java (hunk 1 above)
Comment (CloudNiner) [inline src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java:465]: Ah! I get it! 

Might be worth then placing a comment detailing this at the NihOptions class level, so newcomers are aware of this oddity?

Comment (flibbertigibbet) [inline src/main/java/org/opentripplanner/graph_builder/impl/shapefile/NihShapefileStreetGraphBuilderImpl.java:465]: I was thinking about later changing how this works, or that we might change the shapefile structure to accommodate, but yeah, good idea.

**Pull request merged**